site/models/event.php

```diff
@@ -1,32 +1,37 @@
 <?php
 
-use Kirby\Cms\Page;
 use Kirby\Content\Field;
 use Kirby\Toolkit\Str;
 
-class EventPage extends Page
+class EventPage extends DefaultPage
 {
-	public function date(): Field
+	public function date(string $time = 'start'): Field
 	{
 		return parent::date()->value(
-			$this->num() . ' ' . $this->start() . ' ' . $this->timezone()
+			$this->num() . ' ' . $this->{$time}() . ' ' . $this->timezone()
 		);
 	}
 
-	public function datetime(): DateTime
-	{
+	public function datetime(
+		string|null $timezone = null,
+		string $time = 'start',
+	): DateTime {
 		return new DateTime(
-			(string)$this->date(),
-			new DateTimeZone((string)$this->timezone())
+			datetime: $this->date($time),
+			timezone: new DateTimeZone($timezone ?? $this->timezone())
```

Comment:
I've tested and this doesn't work correct. Still gives local dates. Works with `->setTimezone(new DateTimeZone('UTC'))`.

Reply:
Does it now work for you, @afbora?